### PR TITLE
handle extra exception cases on cleanup and add tests

### PR DESCRIPTION
**Pull request by courtsVII**

### CODE REVIEW

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 1]
```diff
@@ -56,15 +56,19 @@ private void deleteObjectsInBucket(String dataBucket) {
...
-        } else {
-          log.warn("Could not delete object {}", blob);
+        try {
+          boolean suceeded = storage.delete(blob.getBlobId());
+          if (suceeded) {
+            log.info("Deleted object {}", blob);
+          } else {
+            log.warn("Could not delete object {}", blob);
+          }
+        } catch (Exception e) {
```
Comment (massdosage) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: So following on from @patduin's earlier comment, is there a more specific exception type we can catch or do the Google APIs not really do checked exceptions?

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 2]
```diff
@@ -56,15 +56,19 @@ private void deleteObjectsInBucket(String dataBucket) {
...
+          log.warn("Error deleting object {} in bucket {}", blob, dataBucket, e);
         }
       }
     } catch (Exception e) {
-      log.warn("Error deleting objects in bucket {}", dataBucket, e);
+      log.warn("Error fetching objects in bucket {} for deletion", dataBucket, e);
```
Comment (massdosage) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: Same here

File: src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java [hunk 3]
```diff
@@ -133,6 +133,87 @@ public void cleanupWhenDeletionFailsDoesntThrowException() {
...
+  public void cleanupWhenDeletionThrowsExceptionOnBlobDoesntStillCleansRemainingBlobs() {
+    BigQueryExtractionData data = new BigQueryExtractionData();
+    TableId tableId = TableId.of("dataset", "table");
+    when(table.getTableId()).thenReturn(tableId);
+    when(storage.delete(any(BlobId.class))).thenReturn(true);
+    Bucket bucket = mock(Bucket.class);
+    when(bucket.delete()).thenReturn(true);
+    when(storage.get(anyString())).thenReturn(bucket);
+
+    List<Blob> blobs = new ArrayList<>();
```
Comment (massdosage) [inline src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java]: You could make this simpler and just have 3 blobs and throw an exception on the second call to delete() - mockito lets you set up mutliple actions on calls to a mocked method to do this.

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java (hunk 2 above)
Comment (massdosage) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: This case isn't tested, so you need to have your mock throw an exception on `list()` (or `iterateAll()`) and then just check that there are no more interactions on `storage`.

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 4]
```diff
@@ -56,15 +56,19 @@ private void deleteObjectsInBucket(String dataBucket) {
     try {
       Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
```
[the remark this answers is not in the document]
Comment (courtsVII) [reply src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: Yep

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java (hunk 1 above)
Comment (courtsVII) [reply src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: StorageExceptions are thrown but then I can't mock them in the tests as its a final class and I don't want to bring in PowerMock

Comment (patduin) [reply src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: why do you need to mock them in the tests?

**Pull request merged**